### DISCUSSION TITLE
The demo code return a type error

### DISCUSSION THREAD

**Discussion by wxhohomsy** [react-querybuilder/react-querybuilder#614] [Q&A] [answered]
Hello guys, I copy and paste the demo code 
<pre>
import { useState } from 'react';
import type { RuleGroupType } from 'react-querybuilder';
import { QueryBuilder } from 'react-querybuilder';
import { fields } from './fields';
import 'react-querybuilder/dist/query-builder.scss';
import './styles.scss';

const initialQuery: RuleGroupType = { combinator: 'and', rules: [] };

export const App = () =&gt; {
    const [query, setQuery] = useState(initialQuery);

    return (
        &lt;QueryBuilder
            fields={fields}
            query={query}
            onQueryChange={setQuery}
            controlClassnames={{ queryBuilder: 'queryBuilder-branches' }}
        /&gt;
    );
};

</pre>
But it throw type error at `onQueryChange={setQuery}`
<pre>
TS2322: Type Dispatch&lt;SetStateAction&lt;RuleGroupType&gt;&gt; is not assignable to type
(query: RuleGroupType&lt;RuleType&lt;string, string, any, string&gt;, string&gt; | RuleGroupTypeIC&lt;RuleType&lt;string, string, any, string&gt;, string&gt;) =&gt; void
Types of parameters value and query are incompatible.
Type 'RuleGroupType&lt;RuleType&lt;string, string, any, string&gt;, string&gt; | RuleGroupTypeIC&lt;RuleType&lt;string, string, any, string&gt;, string&gt;' is not assignable to type 'SetStateAction&lt;RuleGroupType&gt;'.
Type
RuleGroupTypeIC&lt;RuleType&lt;string, string, any, string&gt;, string&gt;
is not assignable to type SetStateAction&lt;RuleGroupType&gt;
Type
RuleGroupTypeIC&lt;RuleType&lt;string, string, any, string&gt;, string&gt;
is not assignable to type RuleGroupType
Property 'combinator' is missing in type 'RuleGroupTypeIC&lt;RuleType&lt;string, string, any, string&gt;, string&gt;' but required in type '{ combinator: string; rules: RuleGroupArray&lt;RuleGroupType&lt;RuleType&lt;string, string, any, string&gt;, string&gt;, RuleType&lt;string, string, any, string&gt;&gt;; not?: boolean | undefined; }'.
react-querybuilder.d.mts(20, 5): combinator is declared here.
react-querybuilder.d.mts(684, 5): The expected type comes from property onQueryChange which is declared here on type
IntrinsicAttributes &amp; QueryBuilderProps&lt;RuleGroupType&lt;RuleType&lt;string, string, any, string&gt;, string&gt; | RuleGroupTypeIC&lt;RuleType&lt;string, string, any, string&gt;, string&gt;&gt;
</pre>

How should I give correct type

**Comment by jakeboone02** [comment, marked as answer]
This code only satisfies TypeScript in v7, of which there is an alpha release `v7.0.0-alpha.2`. (Alpha 3 should be released tomorrow.)

To get around the type issue in v6, you have to be explicit about how you call the state setter method:

```
// Instead of
onQueryChange={setQuery}

// Use
onQueryChange={q => setQuery(q)}
```

**Reply by wxhohomsy** [reply]
Thanks, I use v7 and it indeed works 😃
